src/ra_machine.erl

```diff
@@ -230,6 +232,14 @@
 
 -callback tick(TimeMs :: milliseconds(), state()) -> effects().
 
+-callback eval_members(ra_server:ra_state(),
```

Comment:
As mentioned below I don't think we need the result parts, also I don't think we need to know if a node was up down, this can all be derived from erlang APIs and isn't necessary to work out if we need to add a member or not. 

Can we reduce the `eval_member` API to just take the current list of members `[ra_server_id()]`, the leader server id server_id() and the MacState?

Reply:
Side note: adding `ra_state()` to the aux API was a mistake, the `ra_state()` is very much an internal concern which is why the aux state stuff isn't really documented. Can't change it now though. RA 3.0 maybe.

Reply:
Correction: I don't even remember what the types are meant to be. :) `ra_state()` is the raft state!

Given this function should only be called by leaders I think it is still not required so can be removed.

Reply:
Done